src/Twig/Extension/CalloutExtension.php

```diff
@@ -42,52 +42,26 @@ public function getFunctions(): array
      * @param Collection|CalloutInterface[] $callouts
      *
      * @return Collection|CalloutInterface[]
-     *
-     * @throws Exception
      */
     public function filterCallouts(Collection $callouts): Collection
     {
         if ($callouts->isEmpty()) {
             return $callouts;
         }
 
-        $now = new DateTime();
-
-        return $callouts->filter(function (CalloutInterface $callout) use ($now) {
-            if (!$callout->isEnabled()) {
-                return false;
-            }
-
-            $start = $callout->getTimePeriodStart();
-            if ($start !== null && $now < $start) {
-                return false;
-            }
-
-            $end = $callout->getTimePeriodEnd();
-            if ($end !== null && $now > $end) {
-                return false;
-            }
-
-            $channelFound = false;
-            foreach ($callout->getChannels() as $channel) {
-                if ($channel->getCode() === $this->channelContext->getChannel()->getCode()) {
-                    $channelFound = true;
-
-                    break;
-                }
-            }
-
-            if (!$channelFound) {
-                return false;
-            }
-
-            return true;
+        return $callouts->filter(function (CalloutInterface $callout) {
+            return $this->renderingCalloutEligibilityChecker->isEligible($callout);
         });
     }
 
     public function calloutClasses(CalloutInterface $callout): string
     {
-        $classes = ['setono-callout', 'setono-callout-code-' . $this->sanitizeClass((string) $callout->getCode())];
+        $classes = [
+            'red', // @todo Add color field to Callout
+            $this->getSemanticUiClassesFromPosition($callout->getPosition()),
```

Comment:
If people don't use Semantic UI for frontend these classes will obscure things

Reply:
Added `CssClassBuilderInterface` for this be have ability to override.